osu.Game/Graphics/UserInterfaceV2/SliderWithTextBoxInput.cs

```diff
@@ -26,10 +27,83 @@ public float KeyboardStep
             set => slider.KeyboardStep = value;
         }
 
+        private readonly BindableNumberWithCurrent<T> current = new BindableNumberWithCurrent<T>();
+
         public Bindable<T> Current
         {
-            get => slider.Current;
-            set => slider.Current = value;
+            get => current;
+            set
+            {
+                current.Current = value;
+                slider.Current = value;
+            }
+        }
+
+        private T? sliderPrecision;
+
+        public T? SliderPrecision
+        {
+            get => sliderPrecision;
+            set
+            {
+                if (value.HasValue)
+                {
+                    T multiple = value.Value / current.Precision;
+                    if (!T.IsNaN(multiple) && !T.IsInfinity(multiple) && !T.IsZero(multiple % T.One))
+                        throw new ArgumentException(@"Precision override must be a multiple of the current precision.");
+                }
+
+                sliderPrecision = value;
+                slider.Current = new BindableNumber<T>
+                {
+                    MinValue = sliderMinValue ?? current.MinValue,
+                    MaxValue = sliderMaxValue ?? current.MaxValue,
+                    Default = current.Default,
+                    Precision = value ?? current.Precision,
+                };
+            }
+        }
+
+        private T? sliderMinValue;
+
+        public T? SliderMinValue
+        {
+            get => sliderMinValue;
+            set
+            {
+                if (value.HasValue && value.Value < current.MinValue)
+                    throw new ArgumentException(@"Minimum value override must be greater than or equal to the current minimum value.");
+
+                sliderMinValue = value;
+                slider.Current = new BindableNumber<T>
+                {
+                    MinValue = value ?? current.MinValue,
+                    MaxValue = sliderMaxValue ?? current.MaxValue,
+                    Default = current.Default,
+                    Precision = sliderPrecision ?? current.Precision,
+                };
+            }
+        }
+
+        private T? sliderMaxValue;
+
+        public T? SliderMaxValue
+        {
+            get => sliderMaxValue;
+            set
+            {
+                if (value.HasValue && value.Value > current.MaxValue)
+                    throw new ArgumentException(@"Maximum value override must be less than or equal to the current maximum value.");
+
+                sliderMaxValue = value;
+                slider.Current = new BindableNumber<T>
+                {
+                    MinValue = sliderMinValue ?? current.MinValue,
+                    MaxValue = value ?? current.MaxValue,
+                    Default = current.Default,
+                    Precision = sliderPrecision ?? current.Precision,
+                };
+            }
```

Comment:
Why is _this_ exposed? Don't you only need precision mutable? What's the point to this?

Reply:
Precision is not the only property in `BindableNumber`, so I added these to match the API capability of my previous version. This is also useful to have in the rotation popover because it allows the slider to be constrained to a range while the text input is unconstrained. This allows the user to input angle values beyond the -360 to 360 range which are still well defined and valid. This could be useful when pasting in values from a source with unnormalized values, like an incremental series.

Reply:
[-360,360] is plenty. I don't want to allow any more than that. Or, at the *least*, it should be getting normalised back to [-360,360].

Reply:
Sure normalising it back to that range is an improvement i guess. Still need the extended range in the text box though because the normalization with period 360 is not gonna happen inside `SliderWithTextBoxInput`, and I can think of other uses for extended range.

Reply:
Can we just start with the decimal precision thing and nix the extended range for now?

Reply:
Sure i guess. I'll split it into separate PR.